packages/web-components/src/components/va-search/va-search.tsx

```diff
@@ -0,0 +1,183 @@
+
+  /**
+   * Fires when the search input gains focus
+   */
+  @Event({
+    composed: true,
+    bubbles: true,
+  })
+  inputFocusEvent: EventEmitter;
+
+  /**
+   * Fires when the search input is focused and a key is pressed
+   */
+  @Event({
+    composed: true,
+    bubbles: true,
+  })
+  inputKeyDownEvent: EventEmitter;
+
+  /**
+   * Fires when the search button is clicked
+   */
+  @Event({
+    composed: true,
+    bubbles: true,
+  })
+  buttonClickEvent: EventEmitter;
+
+  /**
+   * Fires when the search button gains focus
+   */
+  @Event({
+    composed: true,
+    bubbles: true,
+  })
+  buttonFocusEvent: EventEmitter;
+
+  /**
+   * Fires when the search button is focused and a key is pressed
+   */
+  @Event({
+    composed: true,
+    bubbles: true,
+  })
+  buttonKeyDownEvent: EventEmitter;
+
+  /**
+   * Text displayed inside the search button
+   */
+  @Prop() buttonText?: string;
+
+  /**
+   * Hides the search button
+   */
+  @Prop() hideButton: boolean = false;
+
+  /**
+   * Initial value of input
+   */
+  @Prop() inputValue?: string;
+
+  /**
+   * Adds an aria-label attribute to the input and button
+   */
+  @Prop() label: string = 'Search';
+
+  // Input Event Handlers
+  private handleInputBlurEvent = (event: FocusEvent) => {
+    this.inputBlurEvent.emit(event);
+  };
+
+  private handleInputChangeEvent = (event: Event) => {
+    // const target = event.target as HTMLInputElement;
+    this.inputChangeEvent.emit(event);
+  };
+
+  private handleInputFocusEvent = (event: FocusEvent) => {
+    this.inputFocusEvent.emit(event);
+  };
+
+  private handleInputKeyDownEvent = (event: KeyboardEvent) => {
+    this.inputKeyDownEvent.emit(event);
+  };
+
+  // Button Event Handlers
+  private handleButtonClickEvent = (event: KeyboardEvent | MouseEvent) => {
+    this.buttonClickEvent.emit(event);
+  };
+
+  private handleButtonFocusEvent = (event: FocusEvent) => {
+    this.buttonFocusEvent.emit(event);
+  };
+
+  private handleButtonKeyDownEvent = (event: KeyboardEvent) => {
+    this.buttonKeyDownEvent.emit(event);
+  };
+
+  render() {
+    const {
+      buttonText,
+      handleButtonClickEvent,
+      handleButtonFocusEvent,
+      handleButtonKeyDownEvent,
+      handleInputBlurEvent,
+      handleInputChangeEvent,
+      handleInputFocusEvent,
+      handleInputKeyDownEvent,
+      hideButton,
+      inputValue,
+      label,
+    } = this;
+
+    return (
+      <Host>
+        <input
+          id="va-search-input"
+          aria-activedescendant={this.el.getAttribute('aria-activedescendant')}
+          aria-autocomplete="none"
+          aria-controls={this.el.getAttribute('aria-controls')}
+          aria-expanded={this.el.getAttribute('aria-expanded')}
+          aria-haspopup={this.el.getAttribute('aria-haspopup')}
+          aria-label={label}
+          autocomplete="off"
+          onBlur={handleInputBlurEvent}
+          onChange={handleInputChangeEvent}
+          onFocus={handleInputFocusEvent}
```

Comment:
@andresriveratoro should we remove handleInputFocusEvent since we aren't using it in Storybook